moon-store-mac-os/moon-store-mac-os/Networking/Managers/AuthenticationManager.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  AuthenticationManager.swift
+//  moon-store-mac-os
+//
+//  Created by Jose Luna on 1/8/25.
+//
+
+import Foundation
+
+final class AuthenticationManager {
+    private let networkManager: NetworkManager = .init()
+    private let decoder: JSONDecoder = .init()
+    private let userStore: DataManager<UserModel> = .init()
+    
+    var isLoggedUser: Bool {
+        loggedUser != nil
+    }
+    
+    var loggedUser: UserModel? {
+        do {
+            let users = try userStore.fetch()
+            return users.first
+        } catch {
+            debugPrint("Error: User not found \(error.localizedDescription) redirect to login")
+            return nil
+        }
+    }
```

Comment:
Esta variable se hizo muy grando solo por imprimir el error, cosa que no serviria mucho en este caso, podrías resumir este código de la siguiente forma:

```suggestion
    var loggedUser: UserModel? {
            let users = try? userStore.fetch()
            return users?.first
    }
```

Reply:
Se habia dejado asi por feedback anterior. Lo cambiare